src/t8_geometry/t8_geometry_implementations/t8_geometry_examples.cxx

```diff
@@ -349,6 +329,71 @@ t8_geometry_cubed_spherical_shell::t8_geom_evaluate (t8_cmesh_t cmesh, t8_gloidx
   t8_geom_evaluate_sphere_quad_hex (active_tree_vertices, 3, ref_coords, num_coords, out_coords);
 }
 
+void
+t8_geometry_cubed_sphere::t8_geom_evaluate (t8_cmesh_t cmesh, t8_gloidx_t gtreeid, const double *ref_coords,
+                                            const size_t num_coords, double *out_coords) const
+{
+  double n[3]; /* Normal vector. */
+  double r[3]; /* Radial vector. */
+  double s[3]; /* Radial vector for the corrected coordinates. */
+  double p[3]; /* Vector on the plane resp. quad. */
+
+  /* Center hex. */
+  if (gtreeid % 4 == 0) {
+    for (size_t i_coord = 0; i_coord < num_coords; i_coord++) {
+      size_t offset = 3 * i_coord;
+
+      t8_geom_linear_interpolation (ref_coords + offset, active_tree_vertices, 3, 3, p);
+
+      out_coords[offset + 0] = p[0];
+      out_coords[offset + 1] = p[1];
+      out_coords[offset + 2] = p[2];
+    }
+
+    return;
+  }
+
+  n[0] = active_tree_vertices[0 + 0];
+  n[1] = active_tree_vertices[0 + 1];
+  n[2] = active_tree_vertices[0 + 2];
+  t8_vec_normalize (n);
+
+  r[0] = active_tree_vertices[7 * 3 + 0];
+  r[1] = active_tree_vertices[7 * 3 + 1];
+  r[2] = active_tree_vertices[7 * 3 + 2];
```

Comment:
I would love to have some function `get_3d_vertex(active_tree_vertices, 7, r)` to clear up a lot of code, but also probably out of scope of this PR

Reply:
`t8_vec_copy` for the rescue!